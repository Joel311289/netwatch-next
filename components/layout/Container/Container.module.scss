.containerWrapper {
  max-width: var(--size-max);
  width: 100%;
  padding: 0 var(--padding-aside);
  margin: 0 auto;
  background: transparent;
  box-sizing: border-box;
  display: block;
}

.hasMargins {
  margin-top: 15px;
  margin-bottom: 100px;
}
