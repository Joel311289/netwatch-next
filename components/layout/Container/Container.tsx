import { ElementChildren, ElementHTML } from '@/types';
import { classes } from '@/utils/helpers';
import styles from '@/components/layout/Container/Container.module.scss';

export type ContainerProps = typeof defaultProps &
  ElementHTML &
  ElementChildren & {
    margins?: boolean;
  };

const defaultProps = {
  margins: false
};

const Container = ({ children, className, margins }: ContainerProps) => {
  return (
    <div className={classes(styles.wrapper, className, margins && styles.hasMargins)}>
      {children}
    </div>
  );
};

Container.defaultProps = defaultProps;

export default Container;
