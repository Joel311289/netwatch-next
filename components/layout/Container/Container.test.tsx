import { render, screen } from '@testing-library/react';
import Container from '@/components/layout/Container/Container';

describe('Tests Container component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Container>Test</Container>);

    expect(container).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    render(<Container>Test</Container>);

    expect(screen.getByText('Test')).toBeInTheDocument();
  });
});
