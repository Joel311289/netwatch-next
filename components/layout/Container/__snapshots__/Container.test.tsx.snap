// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Container component renders component correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    Test
  </div>
</div>
`;
